src/operations/get_more.ts

```diff
@@ -56,7 +62,40 @@ export class GetMoreOperation extends AbstractOperation {
         new MongoRuntimeError('Getmore must run on the same server operation began on')
       );
     }
-    server.getMore(this.ns, this.cursorId, this.options, callback);
+
+    const collection = this.ns.collection;
+    if (collection == null) {
+      // Cursors should have adopted the namespace returned by MongoDB
+      // which should always defined a collection name (even a pseudo one, ex. db.aggregate())
+      return callback(new MongoRuntimeError('A collection name must be determined before getMore'));
+    }
+
+    const getMoreCmd: GetMoreCommand = {
+      getMore: this.cursorId,
+      collection
+    };
+
+    if (typeof this.options.batchSize === 'number') {
+      getMoreCmd.batchSize = Math.abs(this.options.batchSize);
```

Comment:
There is some *really* weird stuff in the code having to do with handling this, in makeFindCommand specifically. Could you file a ticket to investigate this (for both "limit" and "batchSize") so that we know if this is just confusing legacy stuff or if there's a good reason to do it? If there is a good reason to do it, we can use that ticket to add tests. Otherwise we can use the ticket to delete confusing code.